openml-h2o/src/test/java/com/feedzai/openml/h2o/H2OModelProviderLoadTest.java

```diff
@@ -142,4 +150,54 @@ public DatasetSchema createDatasetSchema(final Set<String> targetValues) {
                         .build()
         );
     }
+
+    /**
+     * Tests that when loading a model with a given field set A that is linked to a given {@link DatasetSchema} B, where both schemas are not compatible,
+     * the loading fails with a {@link ModelLoadingException}.
+     */
+    @Test
+    public final void testLoadModelWithPartialSchema() {
+        final DatasetSchema fullSchema = createDatasetSchema(TARGET_VALUES);
+        final DatasetSchema partialSchema = removeNonTargetVariable(fullSchema);
+
+        final H2OModelCreator modelLoader = getMachineLearningModelLoader(H2OAlgorithm.DISTRIBUTED_RANDOM_FOREST);
+        final String modelPath = this.getClass().getResource("/" + POJO_MODEL_FILE).getPath();
+        Assertions.assertThatThrownBy(() -> modelLoader.loadModel(Paths.get(modelPath), partialSchema))
+                .isInstanceOf(ModelLoadingException.class);
```

Comment:
just to make sure; without your patch, this would fail, right?

Reply:
It did, yes :)